### DISCUSSION TITLE
What are the the best appoaches in designing a chess board

### DISCUSSION THREAD

**Discussion by zferdz** [flet-dev/flet#1464] [Q&A]
### Question

I am specifically interested on the chessboard design. One that is fast in board updating when a piece is captured for example.

Taking the solitaire example, the pieces are the cards, the slots are the board squres. To capture a piece you have to remove the card at tosq slot and move the piece from the fromsq slot to the tosq slot.

The other approach is to build a list of buttons. In each button you add an image of the piece. To update when there is capture, you change the image in the button.

What could be the other possibility?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by InesaFitsner** [comment]
You could try the approach as used in the [ColorPicker](flet_contrib/color_picker/src/color_picker.py).

The board would be a Stack wrapped in GestureDetector (like `color_map_container` in ColorPicker), squares and pieces would be Containers placed in Stack (like `color_map` and `thumb`). You would calculate which piece you picked by coordinates on DragStart event of the board.

**Reply by zferdz** [reply]
The colorpicker link is broken.

**Reply by FeodorFitsner** [reply]
https://github.com/flet-dev/flet-contrib/tree/main/flet_contrib/color_picker

**Comment by zferdz** [comment]
I tried the solitaire approach, but when dragging a piece, the piece image and the square background is also dragged. It should have been the piece only.
